lib/auth/init.go

```diff
@@ -533,6 +533,14 @@ func createPresets(ctx context.Context, asrv *Server) error {
 			if !trace.IsAlreadyExists(err) {
 				return trace.WrapWithMessage(err, "failed to create preset role %v", role.GetName())
 			}
+
+			if role.GetName() == teleport.PresetAuditorRoleName {
+				role = services.UpdateAuditorRoleRFD82(role)
```

Comment:
My understanding is that we should add the `types.KindSessionTracker, RO()` rule if the current version doesn't explicitly allow or deny it.

However, this will pass the `services.NewPresetAuditorRole()` role instead of the cluster's actual role definition.
Doing this will always end up returning the preset and not changing the current version to add the `types.KindSessionTracker, RO()` rule.

I think we should change this to pass the current version as base.

```suggestion
				currentRole, err := asrv.GetRole(ctx, role.GetName())
				if err != nil {
					return trace.Wrap(err)
				}
				role = services.UpdateAuditorRoleRFD82(currentRole)
```

What do you think?

Reply:
I'm doing the same for another resource and was trying to come up with something that could be re-used by future similar needs

It's still WIP but if you want to take a look : https://github.com/gravitational/teleport/pull/16237/files

Reply:
Updated, please take another look now.